File: php/features.php
<?php
declare(strict_types=1);

// Pogony SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PogonyFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PogonyBaseFeature();
            case "test":
                return new PogonyTestFeature();
            default:
                return new PogonyBaseFeature();
        }
    }
}

File: py/features.py
# Pogony SDK feature factory

from feature.base_feature import PogonyBaseFeature
from feature.test_feature import PogonyTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PogonyBaseFeature(),
        "test": lambda: PogonyTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
